src/SFML/Graphics/Shader.cpp

```diff
@@ -786,11 +786,7 @@ bool Shader::compile(const char* vertexShaderCode, const char* geometryShaderCod
     }
 
     // Destroy the shader if it was already created
-    if (m_shaderProgram)
-    {
-        glCheck(GLEXT_glDeleteObject(castToGlHandle(m_shaderProgram)));
-        m_shaderProgram = 0;
-    }
+    m_shaderProgram.reset(0);
```

Comment:
This is taking ownership of a shader program identified with 0, which seems wrong.
Because `m_shaderProgram.reset();` cleans up resource but does not change the stored identifier (which would be bad), I think `m_shaderProgram = {};` is what we need here.